// zrender/src/export.ts
/**
 * Do not mount those modules on 'src/zrender' for better tree shaking.
 */

import * as zrUtil from 'zrender/src/core/util';
import * as matrix from 'zrender/src/core/matrix';
import * as vector from 'zrender/src/core/vector';
import * as colorTool from 'zrender/src/tool/color';
import * as pathTool from 'zrender/src/tool/path';
import {parseSVG} from 'zrender/src/tool/parseSVG';

import * as morphPathTool from 'zrender/src/tool/morphPath';

export {default as Point, PointLike} from 'zrender/src/core/Point';

export {
    default as Element,
    ElementAnimateConfig,
    ElementTextConfig,
    ElementTextGuideLineConfig,
    ElementEvent,
    ElementEventCallback,
    ElementProps
} from 'zrender/src/Element';

export {default as Displayable, DisplayableProps} from 'zrender/src/graphic/Displayable';
export {default as Group, GroupProps} from 'zrender/src/graphic/Group';
export {default as Path, PathStyleProps, PathProps, PathStatePropNames, PathState} from 'zrender/src/graphic/Path';
export {default as Image, ImageStyleProps, ImageProps, ImageState} from 'zrender/src/graphic/Image';
export {default as CompoundPath, CompoundPathShape} from 'zrender/src/graphic/CompoundPath';
export {default as TSpan, TSpanStyleProps, TSpanProps, TSpanState} from 'zrender/src/graphic/TSpan';
export {default as IncrementalDisplayable} from 'zrender/src/graphic/IncrementalDisplayable';
export {default as Text, TextStylePropsPart, TextStyleProps, TextProps, TextState} from 'zrender/src/graphic/Text';

export {default as Arc, ArcProps, ArcShape} from 'zrender/src/graphic/shape/Arc';
export {default as BezierCurve, BezierCurveProps, BezierCurveShape} from 'zrender/src/graphic/shape/BezierCurve';
export {default as Circle, CircleProps, CircleShape} from 'zrender/src/graphic/shape/Circle';
export {default as Droplet, DropletProps, DropletShape} from 'zrender/src/graphic/shape/Droplet';
export {default as Ellipse, EllipseProps, EllipseShape} from 'zrender/src/graphic/shape/Ellipse';
export {default as Heart, HeartProps, HeartShape} from 'zrender/src/graphic/shape/Heart';
export {default as Isogon, IsogonProps, IsogonShape} from 'zrender/src/graphic/shape/Isogon';
export {default as Line, LineProps, LineShape} from 'zrender/src/graphic/shape/Line';
export {default as Polygon, PolygonProps, PolygonShape} from 'zrender/src/graphic/shape/Polygon';
export {default as Polyline, PolylineProps, PolylineShape} from 'zrender/src/graphic/shape/Polyline';
export {default as Rect, RectProps, RectShape} from 'zrender/src/graphic/shape/Rect';
export {default as Ring, RingProps, RingShape} from 'zrender/src/graphic/shape/Ring';
export {default as Rose, RoseProps, RoseShape} from 'zrender/src/graphic/shape/Rose';
export {default as Sector, SectorProps, SectorShape} from 'zrender/src/graphic/shape/Sector';
export {default as Star, StarProps, StarShape} from 'zrender/src/graphic/shape/Star';
export {default as Trochoid, TrochoidProps, TrochoidShape} from 'zrender/src/graphic/shape/Trochoid';

export {default as LinearGradient, LinearGradientObject} from 'zrender/src/graphic/LinearGradient';
export {default as RadialGradient, RadialGradientObject} from 'zrender/src/graphic/RadialGradient';
export {
    default as Pattern,
    PatternObjectBase,
    PatternObject,
    ImagePatternObject,
    SVGPatternObject
} from 'zrender/src/graphic/Pattern';
export {default as BoundingRect, RectLike} from 'zrender/src/core/BoundingRect';
export {default as OrientedBoundingRect} from 'zrender/src/core/OrientedBoundingRect';

export {matrix};
export {vector};
export {colorTool as color};
export {pathTool as path};
export {zrUtil as util};

export {morphPathTool as morph};

export {parseSVG};

export {default as showDebugDirtyRect} from 'zrender/src/debug/showDebugDirtyRect';

export {setPlatformAPI} from 'zrender/src/core/platform';
